packages/substrate-client/src/chainhead/storage.ts

```diff
@@ -20,45 +21,83 @@ export const createStorageFn = (
     | OperationStorageDone
     | OperationWaitingForContinue
   >,
-) => {
+): FollowResponse["storage"] => {
   const cbStore = createStorageCb(request)
-  return abortablePromiseFn(
-    <Type extends StorageItemInput["type"]>(
-      resolve: (value: StorageResult<Type>) => void,
-      reject: (e: Error) => void,
-      hash: string,
-      type: Type,
-      key: string,
-      childTrie: string | null,
-    ) => {
-      const isDescendants = type.startsWith("descendants")
-      let result: any = isDescendants ? [] : null
+  return abortablePromiseFn((resolve, reject, hash, type, key, childTrie) => {
+    const result: {
+      value: StorageResult<"value">
+      hash: StorageResult<"hash">
+      closestDescendantMerkleValue: StorageResult<"closestDescendantMerkleValue">
+      descendantsValues: StorageResult<"descendantsValues">
+      descendantsHashes: StorageResult<"descendantsHashes">
+    } = {
+      value: null,
+      hash: null,
+      closestDescendantMerkleValue: null,
+      descendantsValues: [],
+      descendantsHashes: [],
+    }
+    const onItems = (items: StorageItemResponse[]) => {
+      if (isDescendantsValues(type)) {
+        result[type].push(...(items as Array<{ key: string; value: string }>))
```

Comment:
here we are casting but i think its ok since it must be defined based on the spec. and if the JSON-RPC server for whatever reason returns `undefined` for this key, its not our problem because the spec says it must be defined